src/core/thread/mle_tlvs.hpp

```diff
@@ -1292,10 +1292,18 @@ class ConnectivityTlv : public Tlv
      * @retval FALSE  If the TLV does not appear to be well-formed.
      *
      */
-    bool IsValid(void) const
+    bool IsValid(void) const { return (GetLength() == sizeof(*this) - sizeof(Tlv) || IsValidDefaultSedCapacity()); }
+
+    /**
+     * This method indicates whether or not the sed buffer size and datagram count are included.
+     *
+     * @retval TRUE   If the sed buffer size and datagram count are included.
+     * @retval FALSE  If the sed buffer size and datagram count are not included.
+     *
+     */
+    bool IsValidDefaultSedCapacity(void) const
     {
-        return (GetLength() == sizeof(*this) - sizeof(Tlv) ||
-                GetLength() == sizeof(*this) - sizeof(Tlv) - sizeof(mSedBufferSize) - sizeof(mSedDatagramCount));
+        return GetLength() == sizeof(*this) - sizeof(Tlv) - sizeof(mSedBufferSize) - sizeof(mSedDatagramCount);
```

Comment:
This implementation does not match the comment above. How about renaming and changing implementation to:
```c++
bool IsSedBufferingIncluded(void) const
{
    return GetLength() >= sizeof(*this) - sizeof(Tlv);
}
```
You will also have to negate the associated calls.